### ISSUE TITLE
Own dataset

### ISSUE THREAD

**Issue by PoojaAxis** [ParisNeo/Gpt4All-webui#21]
How to use our own data?

**Comment by ParisNeo** [comment]
Yes, that's possible, We didn't add this to our ui yet. You'll have to wait or do that using the old way with console, yaml confuiguration files and stuff.

You can do all that from the main repo gpt4all:
https://github.com/nomic-ai/gpt4all

you have a train script to retrain with your own data

**Comment by ParisNeo** [comment]
Well, reading pdfs will be handled by an extension. Just need to wait a little bit.
You can simply copy manually text and place it in the discussion then ask questions for now. Or even create a personality that contains the text in the conditionning.

The main issue here is context size. The onctext is not big enough to read long text. That's a current limitation that may be fixed when Recurrent Transformers become main stream.

**Comment by PoojaAxis** [comment]
Good Day!

Well what do you suggest if we were to upload all of the PDF files onto the database? Or
import the contents of all the PDF files into the database?

Regards,



***@***.***
***@***.***
Pooja Ramdowor

***@***.***2nd Floor, The AXIS, 26 Bank Street, Cybercity, Ebene 72201, Mauritius
***@***.*** (+230) 403 2500 | ***@***.***   (+230) 403 2501 | ***@***.***  www.axis.mu<https://www.axis.mu>
Offices: Mauritius | Kenya | Seychelles | South Africa

***@***.***
BRN: C08082328
         Follow us on
***@***.***<https://www.linkedin.com/company/2034354?trk=tyah&trkInfo=tarId%3A1405419839500%2Ctas%3Aaxis%20fidu%2Cidx%3A2-1-2>
***@***.***<https://www.facebook.com/AxisFiduciaryLtd>
***@***.***<https://twitter.com/AxisFiduciary>

***@***.***
Axis Fiduciary Ltd is licensed by the Financial Services Commission of Mauritius.
This email and all its contents are subject to the following disclaimer<https://www.axis.mu/legal-copyright/>
***@***.***<https://www.axis.mu/>

**Comment by ParisNeo** [comment]
It is possible to do lora fine tuning of your model using a pdf database that you convert to text somehow. But keep in mind that you need to have discussion format, or at least have some examples that resembles a discussion about a text so you don't make the model loose its conversational capabilities. Then you can play on parameters.to adjudt how much of each stuff you want to use. All an exercice of balance.
